File: src/main/java/no/ntnu/webdev/webproject7/user/UserEntity.java
package no.ntnu.webdev.webproject7.user;

import no.ntnu.webdev.webproject7.crud.CrudModel;

import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class UserEntity implements CrudModel<String> {
    @Id
    private String id;
    private Boolean admin;
    private String email;
    private String password;

    public UserEntity(String id, Boolean admin, String email, String password) {
        this.id = id;
        this.admin = admin;
        this.email = email;
        this.password = password;
    }

    protected UserEntity() {}

    public String getId() {
        return this.id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Boolean getAdmin() {
        return this.admin;
    }

    public void setAdmin(Boolean admin) {
        this.admin = admin;
    }

    public String getEmail() {
        return this.email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return this.password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: src/main/java/no/ntnu/webdev/webproject7/user/UserEntity.kt
package no.ntnu.webdev.webproject7.user

import no.ntnu.webdev.webproject7.crud.CrudModel
import javax.persistence.Entity
import javax.persistence.Id

typealias UserEntityId = String;

@Entity
class UserEntity : CrudModel<UserEntityId> {
    @Id
    override var id: UserEntityId? = null
    var admin: Boolean? = null
    var email: String? = null
    var password: String? = null

    constructor(id: UserEntityId?, admin: Boolean?, email: String?, password: String?) {
        this.id = id
        this.admin = admin
        this.email = email
        this.password = password
    }

    protected constructor()
}
